orca_detector/generator.py

```diff
@@ -1,94 +1,180 @@
+"""
+Keras custom generators for the Orca project.
+
+W251 (Summer 2019) - Spyros Garyfallos, Ram Iyer, Mike Winton
+
+Adapted from https://github.com/DTaoo/VGGish
+"""
+
 import keras
 import librosa
 import numpy as np
+import pprint
 import resampy
 import soundfile as sf
 
-import params
-from vggish.mel_features import frame, log_mel_spectrogram
-
+# project-specific imports
+import orca_params
+import vggish_params as params
+from mel_features import frame, log_mel_spectrogram
 
 def _waveform_to_examples(data, sample_rate):
```

Comment:
I'm not sure about the examples part of this method name, maybe something like _waveform_to_mel_spectrogram_segments

Reply:
Agreed, I like that better.  It took me a while to understand how a single sample was becoming multiple "examples".